### ISSUE TITLE
cowrie config and port enable/disable 

### ISSUE THREAD

**Issue by smartclientt** [dtag-dev-sec/tpotce#649]
Hello, i have two question 
-after changing for example cowrie configuration file, hwo i enable this change?
-how i can disable some port from my T-pot to not be scaned for example with Nmap?
thanks and have a good day

**Comment by pisces-period** [comment]
Hi @smartclientt, 

TL;DR:

__after changing for example cowrie configuration file, how i enable this change?__

Edit the docker-compose.yaml file you are using to spawn up the docker containers and mount your custom configuration file as a volume. 

You can achieve this by adding the following line to your docker-compose.yaml:

```
cowrie:
    build: .
    container_name: cowrie
    restart: always
    tmpfs:
     - /tmp/cowrie:uid=2000,gid=2000
     - /tmp/cowrie/data:uid=2000,gid=2000
    networks:
     - cowrie_local
    ports:
     - "22:22"
     - "23:23"
    image: "dtagdevsec/cowrie:2006"
    read_only: true
    volumes:
     - /data/cowrie/downloads:/home/cowrie/cowrie/dl
     - /data/cowrie/keys:/home/cowrie/cowrie/etc
     - /data/cowrie/log:/home/cowrie/cowrie/log
     - /data/cowrie/log/tty:/home/cowrie/cowrie/log/tty
     - ${/path/to/cowrie.cfg}:/home/cowrie/cowrie/cowrie.cfg
```

Where:

- '${/path/to/cowrie.cfg}`: absolute path of your custom cowrie.cfg file on your local machine (INCLUDING the file name)

You'd then need to re-deploy the cowrie service, so that the container indeed receives the new configuration file.

__how i can disable some port from my T-pot to not be scanned for example with nmap?__

The answer here is, it depends. 

The cleanest way would be to stop the container/process that is occupying the port you wish to disable.

Alternatively you can use iptables. Something like the following command should work:

`sudo iptables -A INPUT -p tcp --dport ${port_number} -m state --state NEW,ESTABLISHED -j DROP`

However this means you're dropping inbound traffic to a service which is supposed to be exploitable and then again, why would you have the service running in this case? ¯\\\_(ツ)\_/¯

The best option is to simply stop the containers that are using the port you wish to disable.

**Comment by smartclientt** [comment]
**thank you for this help**

**Comment by t3chn0m4g3** [comment]
The `/opt/tpot/bin/rules.sh` script extracts all port infos from `/opt/tpot/etc/tpot.yml`. You need to make the script aware of custom changes for the iptables ruleset or your installation will probably run into connectivity problems in the long run.
Im guessing the question is answered and the issue resolved. Closing this.
